### ISSUE TITLE
Different traceback behaviour in chameleon ZPT implementation

### ISSUE THREAD

**Issue by tomgross** [malthe/chameleon#167]
I have the following use case: I have a snippet which renders Google Analytics code in a template. I used to look into sys.exc_info() to see if an exception was raised. This worked perfectly for zope.pagetemplate. See the test:

```
def test_tb(self):
    import sys
    from zope.pagetemplate.pagetemplate import PageTemplate
    t = PageTemplate()
    t.write('<div tal:content="y|nothing"></div><span tal:content="options/error" />')
    printerror = lambda:sys.exc_info()
    got = t(error=printerror)
    self.assertTrue('<span>(None, None, None)</span>' in got)
```

For Chameleon this somehow does not work. The traceback information which is swallowed by the pipe in the zope implementation is not in the Chameleon implementation of pagetemplates. Here is the same test failing with a NameError('y') for Chameleon: 

```
def test_tb(self):
    import sys
    from chameleon.zpt.template import PageTemplate
    template = PageTemplate('<div tal:content="y|nothing"></div><span tal:content="error()" />')
    printerror = lambda:sys.exc_info()
    got = template.render(error=printerror)
    self.assertTrue('<span>(None, None, None)</span>' in got)
```

I consider this a bug. Any pointers how to fix, workaround this?

**Comment by malthe** [comment]
Fixed in 9912882.

**Comment by tomgross** [comment]
Thanks!
